### PR TITLE
ec2_vpc_nat_gateway: Add support for connectivity_type = private

### PR DESCRIPTION
**Pull request by GomathiselviS**
Signed-off-by: GomathiselviS <gomathiselvi@gmail.com>

##### SUMMARY

Fixes: #1260 

This PR adds a key 'connectivity_type' to ec2_vpc_nat_gateway module

##### ISSUE TYPE

- Feature Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: tests/integration/targets/ec2_vpc_nat_gateway/tasks/main.yml [hunk 1]
```diff
@@ -881,6 +882,51 @@
...
+      subnet_id: '{{ subnet_id }}'
+      connectivity_type: 'private'
+      wait: yes
+    register: create_ngw
+    check_mode: yes
+
+  - name: Assert creation happened (expected changed=true) - CHECK_MODE
+    assert:
+      that:
+      - create_ngw.changed
```
Comment (mandar242) [inline tests/integration/targets/ec2_vpc_nat_gateway/tasks/main.yml:906]: overall LGTM! nit. for `check_mode` assertions, I like to add assertion for resource_actions to make sure it did not really create the resource.
so something like, but up to you.
`- '"ec2:CreateNatGateway" not in create_ngw.resource_actions'`

**Review by mandar242 — APPROVED** [review]

File: plugins/modules/ec2_vpc_nat_gateway.py [hunk 2]
```diff
@@ -614,6 +624,9 @@ def create(client, module, subnet_id, allocation_id, tags, client_token=None,
         changed = True
         return changed, result, msg
 
+    if connectivity_type == "private":
+        params.pop('AllocationId')
```
Comment (goneri) [inline plugins/modules/ec2_vpc_nat_gateway.py:628]: The fact we set the `AllocationId` key and we delete it 20 lines below is a bit inelegant.

File: plugins/modules/ec2_vpc_nat_gateway.py [hunk 3]
```diff
@@ -544,13 +552,14 @@ def release_address(client, module, allocation_id):
 
 
 def create(client, module, subnet_id, allocation_id, tags, client_token=None,
```
Comment (goneri) [inline plugins/modules/ec2_vpc_nat_gateway.py:554]: I could be nice to cover the `params` dict creation with a unit-tests.

**Review by goneri — APPROVED** [review]

File: plugins/modules/ec2_vpc_nat_gateway.py (hunk 3 above)
Comment (GomathiselviS) [reply plugins/modules/ec2_vpc_nat_gateway.py:554]: I would like to create a separate PR to add unit test cases for the complete module.

**Pull request merged**